StreamNet/Producers/Publisher.cs

```diff
@@ -1,28 +1,60 @@
 using System;
 using System.Threading.Tasks;
 using Confluent.Kafka;
+using Confluent.SchemaRegistry;
+using Confluent.SchemaRegistry.Serdes;
+using StreamNet.Extensions.Encoders;
 using StreamNet.Serializers;
 
 namespace StreamNet.Producers
 {
     public class Publisher : IPublisher
     {
-        public Publisher()
+        public Publisher() => Settings.GetInstance();
+
+        public async Task ProduceAvroAsync<T>(T message, string? topicName = null)
```

Comment:
Do you really need `ProduceAvroAsync` to be `async` itself? 

I explain: the task responsible for serialize and dispatch the message to the topic already runs `async` (even awaiting for it)... 

Maybe you could remove the `async` here and continue without `await` on L19, that way the publisher method will synchronously envelop the producing action in an async task. That make sense?